### ISSUE TITLE
WP Engine - E_WARNING on RecursiveDirectoryIterator when clearing cache

### ISSUE THREAD

**Issue by alfonso100** [wp-media/wp-rocket#4040]
**Before submitting an issue please check that you’ve completed the following steps:**
YES - Made sure you’re on the latest version
YES - Used the search feature to ensure that the bug hasn’t been reported before

**Describe the bug**
On a website hosted in WP Engine, after clearing the cache using any of our Clear Cache functions, the minified files are indeed being cleared, however, for some reason, New Relic catches this **E_WARNING**:

```
 in RecursiveDirectoryIterator::__construct called at ? (?)
 in RecursiveDirectoryIterator::getChildren called at /nas/content/live/xxx/wp-content/plugins/wp-rocket/inc/functions/files.php (382)
 in rocket_clean_minify called at /nas/content/live/xxx/wp-content/plugins/wp-rocket/inc/common/purge.php (439)
 in do_admin_post_rocket_purge_cache called at /nas/content/live/xxx/wp-includes/class-wp-hook.php (292)
 in WP_Hook::apply_filters called at /nas/content/live/xxx/wp-includes/class-wp-hook.php (316)
 in WP_Hook::do_action called at /nas/content/live/xxx/wp-includes/plugin.php (484)
 in do_action called at /nas/content/live/xxx/wp-admin/admin-post.php (70)
```

I've been loggin the **inc/functions/files.php** in several places. 
$iterator is empty here:
https://github.com/wp-media/wp-rocket/blob/trunk/inc/functions/files.php#L336

**To Reproduce**
1. Clear WP Rocket's cache
2. See the E_WARNING logged in new relic
3. Verify that $iterator is empty here: https://github.com/wp-media/wp-rocket/blob/48f5466c8e9699f23318968a943b1302abadcab6/inc/functions/files.php#L336
4. Verify the min files are still cleared

**Expected behavior**
There should be no warning when clearing the cache

**Additional context**
Ticket: https://secure.helpscout.net/conversation/1518038877/265209/

**Backlog Grooming (for WP Media dev team use only)**
- [ ] Reproduce the problem
- [ ] Identify the root cause
- [ ] Scope a solution
- [ ] Estimate the effort

**Comment by webtrainingwheels** [comment]
Same issue is happening here: https://secure.helpscout.net/conversation/1968230782/359797?folderId=377611
Also a WP Engine site. In this case the warning is related to the cache/busting folder:

`RecursiveDirectoryIterator::__construct(/nas/content/live/xxx/wp-content/cache/busting/1/): failed to open dir: No such file or directory`

`in RecursiveDirectoryIterator::__construct called at /nas/content/live/xx/wp-content/plugins/wp-rocket/inc/functions/files.php (1268)
in _rocket_get_cache_path_iterator called at /nas/content/live/wp-content/plugins/wp-rocket/inc/functions/files.php (415)
in rocket_clean_cache_busting called at /nas/content/live/wp-content/plugins/wp-rocket/inc/common/purge.php (442)
in do_admin_post_rocket_purge_cache called at /nas/content/live/wp-includes/class-wp-hook.php (307)
in WP_Hook::apply_filters called at /nas/content/wp-includes/class-wp-hook.php (331)
in WP_Hook::do_action called at /nas/content/wp-includes/plugin.php (476)
in do_action called at /nas/content/wp-admin/admin-post.php (85)`

**Comment by php4fan** [comment]
Still not fixed after 1.5 years, and you charge for this c**p?

**Comment by piotrbak** [comment]
There's no feedback from customers regarding this one. It's not going to be implemented in the near future. We're open to discuss and reopen the issue though.
